File: src/main/java/ar/edu/um/haberes/rest/model/view/LegajoCursoCantidad.java
/**
 * 
 */
package ar.edu.um.haberes.rest.model.view;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import org.hibernate.annotations.Immutable;

import lombok.Data;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Immutable
@Table(name = "vw_legajo_curso_cantidad")
public class LegajoCursoCantidad implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 3357658398144165069L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private String uniqueId;

	private Integer anho;
	private Integer mes;
	private Long legajoId;
	private Integer facultadId;
	private Integer geograficaId;
	private Integer anuales;
	private Integer semestre1;
	private Integer semestre2;

}

File: src/main/java/um/haberes/rest/kotlin/model/view/LegajoCursoCantidad.kt
package um.haberes.rest.kotlin.model.view

import jakarta.persistence.*
import org.hibernate.annotations.Immutable

@Entity
@Table(name = "vw_legajo_curso_cantidad")
@Immutable
data class LegajoCursoCantidad(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var uniqueId: String? = null,

    var anho: Int? = null,
    var mes: Int? = null,
    var legajoId: Long? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var anuales: Int? = null,
    var semestre1: Int? = null,
    var semestre2: Int? = null

)
